README.md

```diff
@@ -40,4 +60,42 @@ $ cd build.release
 $ cmake .. -DCMAKE_TOOLCHAIN_FILE=../submodules/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_BUILD_TYPE=Release -DUSE_CPPRESTSDK=true
 $ cmake --build . --config Release
 ```
-Output will be in `build.release/bin/`
\ No newline at end of file
+Output will be in `build.release/bin/`
+
+## Example usage
+
+```cpp
+#include "signalrclient/hub_connection.h"
+#include "signalrclient/hub_connection_builder.h"
+#include "signalrclient/signalr_value.h"
+
+std::promise<void> start_task;
+signalr::hub_connection connection = signalr::hub_connection_builder::create("http://localhost:5000/hub").build();
+
+connection.on("Echo", [](const signalr::value& m)
+{
+    std::cout << m.as_array()[0].as_string() << std::endl;
+});
+
+connection.start([&start_task](std::exception_ptr exception) {
+    start_task.set_value();
+});
+
+start_task.get_future().get();
+
+std::promise<void> send_task;
+std::vector<signalr::value> arr { "Hello world" };
+signalr::value arg(arr);
+connection.invoke("Echo", arg, [&send_task](const signalr::value& value, std::exception_ptr exception) {
+    send_task.set_value();
+});
+
+send_task.get_future().get();
+
+std::promise<void> stop_task;
+connection.stop([&stop_task](std::exception_ptr exception) {
+    stop_task.set_value();
+});
+
+stop_task.get_future().get();
```

Comment:
I tried running the sample with a minimal CMakeLists.txt:

```
cmake_minimum_required (VERSION 3.5)
project (signalrclient-sample)

find_path(SIGNALR_INCLUDE_DIR signalrclient/hub_connection.h)
include_directories(${SIGNALR_INCLUDE_DIR})

add_executable (sample sample.cpp)
```

But I got the following error:

```
C:\dev\halter73\SignalR-Client-Cpp-Sample\build❯ cmake --build .
Microsoft (R) Build Engine version 16.9.0-preview-20570-04+b04541221 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/dev/halter73/SignalR-Client-Cpp-Sample/CMakeLists.txt
  sample.cpp
C:\dev\microsoft\vcpkg\installed\x64-windows\include\signalrclient\log_writer.h(15,57): error C2039: 'string': is not a
 member of 'std' [C:\dev\halter73\SignalR-Client-Cpp-Sample\build\sample.vcxproj]
...
```

After adding `#include <iostream>` above the `#include "signalrclient/hub_connection.h"`, there was a similar error complaining `'promise': is not a member of 'std'` which was fixed by adding `#include <future>`. I think that the signalr client should be including these automatically if they're not already included.

Now with the header issues fixed, I'm getting linker errors:

```
C:\dev\halter73\SignalR-Client-Cpp-Sample\build❯ cmake --build .
Microsoft (R) Build Engine version 16.9.0-preview-20570-04+b04541221 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Building Custom Rule C:/dev/halter73/SignalR-Client-Cpp-Sample/CMakeLists.txt
  sample.cpp
sample.obj : error LNK2019: unresolved external symbol "public: __cdecl signalr::value::value(char const *)" (??0value@
...
```

I'm definitely no cmake expert, so I'm guessing I'm missing something in my CMakeLists.txt. Do I need to call `find_package()` or `find_library()`? I couldn't come up with a variation of these that worked. It would be nice if the example usage included a minimal CMakeLists.txt. I pushed my broken sample to https://github.com/halter73/SignalR-Client-Cpp-Sample

Reply:
I think figured out the right `find_libary()` command to get the linker working. I needed to add the following:

```
find_library(SIGNALR_LIBRARY NAMES signalrclient PATHS {SIGNALR_INCLUDE_DIR} REQUIRED)
link_libraries(${SIGNALR_LIBRARY})
```

I've updated the 

Everything seems to work now. I think including the normal CMakeList.txt commands required to include the signalrclient headers and link the signalrclient library in the README would be helpful. I pushed the fixed CMakeList.txt to my sample repo.